### PR TITLE
genswagger: don't emit default response if there's no Error definition

### LINKED ISSUE #1162: Swagger definition is broken if there's no default error response set
Introduced by #1127 , refs [utrack/clay#81](https://github.com/utrack/clay/issues/81)

Right now gen-swagger generates broken Swagger def if there's no `.grpc.gateway.runtime.Error` set ([lookup code](https://github.com/grpc-ecosystem/grpc-gateway/pull/1127/files#diff-d419038d63db0fdabc61f0d6cc6683aeR1051)).
Every handler has a link to error definition as "default" response, even if `Error` was not discovered. 

I guess it's better to drop error response definition if `Error` was not found, but still emit a warning (`no message found: Error`). What do you think?
I can send a PR with fix if you're OK with that.

Also, the message can be made better (in fact, generator tries to resolve `.grpc.gateway.runtime.Error`, not just `Error`).

### PR DESCRIPTION
**Pull request by utrack**
Swagger: do not create default response definitions if we can't find Error/StreamError in Registry, even if `DisableDefaultErrors()` is false.

Logs are still emitted if default errors are enabled, but error definition was not found.

grep: `.grpc.gateway.runtime.Error`, `.grpc.gateway.runtime.StreamError`

Closes #1162

### CODE REVIEW

File: protoc-gen-swagger/genswagger/template_test.go [hunk 1]
```diff
@@ -2074,3 +2075,86 @@ func TestMessageOptionsWithGoTemplate(t *testing.T) {
...
+	}
+
+	defRsp, ok := result.Paths["/v1/echo"].Post.Responses["default"]
+	if !ok {
+		return
+	}
+
+	ref := defRsp.Schema.schemaCore.Ref
+	// TODO utrack: is there a better way to find object by ref?
+	refName := strings.TrimPrefix(ref, "#/definitions/")
```
Comment (utrack) [inline protoc-gen-swagger/genswagger/template_test.go]: I'm not sure if there's any existing helper function to get a definition by its ref? strings.TrimPrefix looks ugly

File: protoc-gen-swagger/genswagger/template.go [hunk 2]
```diff
@@ -518,14 +518,24 @@ func renderEnumerationsAsDefinition(enums enumMap, d swaggerDefinitionsObject, r
 
 // Take in a FQMN or FQEN and return a swagger safe version of the FQMN
 func fullyQualifiedNameToSwaggerName(fqn string, reg *descriptor.Registry) string {
+	ret, _ := fullyQualifiedNameToSwaggerNameOk(fqn, reg)
+	return ret
+}
+
+// Take in a FQMN or FQEN and return a swagger safe version of the FQMN and
+// a boolean indicating if FQMN was properly resolved.
+// TODO utrack: bad name?
```
Comment (johanbrandhorst) [inline protoc-gen-swagger/genswagger/template.go]: Just remove the other one and use the one that reports if it was found everywhere. It's a good change :+1:.

File: protoc-gen-swagger/genswagger/template_test.go (hunk 1 above)
Comment (johanbrandhorst) [reply protoc-gen-swagger/genswagger/template_test.go]: Haha not that I know :joy:. This is fine. Remove the comment.

**Comment by utrack** [comment]
@johanbrandhorst the code looks a bit messier now because of more if branches; however, I don't want to do refactoring in the same PR so the changes would be as small as possible. This calls for a separate PR :)

**Comment by johanbrandhorst** [comment]
Would appreciate a further refactor in a separate PR if you have the time :).

**Pull request merged**